File: solutions/julia/hamming/1/hamming.jl
"Calculates the Hamming Distance between two DNA strands."
function distance(a, b)
    if length(a) != length(b)
        throw(ArgumentError(""))
    end
    
    count = 0
    for i = 1:length(a)[1]
        if a[i] != b[i]
            count += 1
        end
    end
    return count
end

File: solutions/php/hamming/1/Hamming.php
<?php

declare(strict_types=1);

function distance(string $strandA, string $strandB): int {

    if ( strlen($strandA) != strlen($strandB) ) {
        $err_msg = "DNA strands must be of equal length.";
        throw new InvalidArgumentException($err_msg);
    }

    $arrayA = str_split($strandA);
    $arrayB = str_split($strandB);
    $count = 0;
    for ( $i = 0; $i < count($arrayA); $i++ ) {
        if ( $arrayA[$i] != $arrayB[$i] ) $count++;
    }

    return $count;
  }
